R/setup_docs.R

```diff
@@ -0,0 +1,104 @@
+#' Initialize documentation website settings
+#'
+#' @description
+#' Creates a subdirectory called `altdoc/` in the package root directory to store the settings files used to by one of the documentation generator utilities (docsify, docute, or mkdocs). The files in this folder are never altered automatically by `altdoc` unless the user explicitly calls `overwrite=TRUE`. They can thus be edited manually to customize the sidebar and website.
+#'
+#' @param tool String. "docsify", "docute", or "mkdocs".
+#' @param path Path to the package root directory.
+#' @param overwrite Logical. If TRUE, overwrite existing files. Warning: This will completely delete the settings files in the `altdocs` directory, including any customizations you may have made.
+#'
+#' @export
+#'
+#' @return No value returned.
+#' 
+#' @examples
+#' if (interactive()) {
+#' 
+#'   # Create docute documentation
+#'   setup_docs(tool = "docute")
+#'
+#'   # Create docsify documentation
+#'   setup_docs(tool = "docsify")
+#'
+#'   # Create mkdocs documentation
+#'   setup_docs(tool = "mkdocs")
+#' 
+#' }
+setup_docs <- function(tool, path = ".", overwrite = FALSE) {
+
+  safe.copy <- function(src, tar, overwrite) {
+    if (fs::file_exists(tar) && !isTRUE(overwrite)) {
+      cli::cli_abort("{tar} already exists. Delete it or set `overwrite=TRUE`.", call = NULL)
+    } else {
+      file.copy(src, tar, overwrite = TRUE)
+    }
+  }
+
+  # input sanity checks
+  if (missing(tool) ||
+      !is.character(tool) ||
+      length(tool) != 1 ||
+      !tool %in% c("docute", "docsify", "mkdocs")) {
+    cli::cli_abort(
+      'The `tool` argument must be "docsify", "docute", or "mkdocs".')
+  }
+
+  # paths
+  path <- .convert_path(path)
+  .check_is_package(path)
+  altdoc_dir <- fs::path_abs(fs::path_join(c(path, "altdoc")))
+  docs_dir <- fs::path_abs(fs::path_join(c(path, "docs")))
+
+  if (!fs::dir_exists(altdoc_dir)) {
+    cli::cli_alert_info("Creating `altdoc/` directory.")
+    fs::dir_create(altdoc_dir)
+  } else if (isTRUE(overwrite)) {
+    try(fs::file_delete(fs::path_join(c(altdoc_dir, "mkdocs.yml"))), silent = TRUE)
```

Comment:
shouldn't we just delete the `atldoc` dir and recreate it?

Reply:
No, because we want users to be able to store permanent static files (images, custom html, javascript) in this folder, and have them be copied over to `docs/`. This is a really powerful customization mechanism.